sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/PublishFailedEventArgs.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Messaging.EventHubs.Producer
+{
+    /// <summary>
+    /// This class allows the application and the producer to pass contextual information
+    /// back and forth on the event of a batch failing to be sent to the consumer successfully.
+    /// </summary>
+    public class PublishFailedEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Events is an IEnumerable of the events in the EventDataBatch that failed to send.
+        /// </summary>
+        public IEnumerable<EventData> Events
+        {
+            get
+            {
+                //this.ThrowifDisposed?
+                return this.events;
+            }
+        }
```

Comment:
```suggestion
        public IEnumerable<EventData> Events { get; }
```

You can use the short form and assign in the constructor.   No need for the explicit backing store.